app/src/main/java/com/mybus/activity/MainActivity.java

```diff
@@ -763,26 +766,41 @@ public void onSearchButtonClick() {
     public void onInfoWindowClick(final Marker marker) {
         //Some infoWindow was clicked
         MyBusMarker myBusMarker = null;
-        //Only matters the Start or End location markers
+        //Detect which type of marker is:
         if ((mStartLocationMarker.getMapMarker() != null) && (mStartLocationMarker.getMapMarker().getId().equals(marker.getId()))) {
             myBusMarker = mStartLocationMarker;
         } else if ((mEndLocationMarker.getMapMarker() != null) && (mEndLocationMarker.getMapMarker().getId().equals(marker.getId()))) {
             myBusMarker = mEndLocationMarker;
-        }
+        } else if (mUserLocationMarker.getMapMarker() != null && (mUserLocationMarker.getMapMarker().getId().equals(marker.getId()))) {
```

Comment:
Looking at the code. There is already a method doing that. 

>     /**
     * Checks if the given marker is StartLocation, EndLocation or other.
     *
     * @param marker
     * @return a MyBusMarker (StartLocation/EndLocation) or null
     */
    public MyBusMarker isMarkerPresent(Marker marker) {
        if (mStartLocationMarker != null && mStartLocationMarker.getMapMarker() != null && mStartLocationMarker.getMapMarker().getId().equals(marker.getId())) {
            return mStartLocationMarker;
        }
        if (mEndLocationMarker != null && mEndLocationMarker.getMapMarker() != null && mEndLocationMarker.getMapMarker().getId().equals(marker.getId())) {
            return mEndLocationMarker;
        }
        return null;
    }